### PR TITLE
Improve KISSConfig

### PR DESCRIPTION
**Pull request by markuspi**
## Problem

When specifying a custom YAML config, all config values needed to be specified. There was no fallback behavior to selectively override values.

## Solution

This PR introduces `pydantic` to validate and manage the configration.
The default values are defined in `config.py`.
YAML config files can override as many or as few items as needed.
Cleaned up generated yaml file: Output file is plain YAML without `!!python/object/new:easydict.EasyDict`

## Changes

* Remove `easydict` dependency
* Add `pydantic` dependency
* Change dataclass-based config to pydantic based config
* Add custom loader for YAML
* Remove default yaml path from CLI. Default values are now retrieved directly from config.py.
* Rename `default.yaml` to `basic.yaml` to avoid confusion. Editing this file only has effect when it is referenced via `--config`!

### CODE REVIEW

File: src/python/kiss_icp/pipeline.py [hunk 1]
```diff
@@ -42,9 +41,7 @@ class OdometryPipeline:
     def __init__(
         self,
         dataset,
-        config: Path,
-        deskew: Optional[bool] = False,
-        max_range: Optional[float] = None,
+        config: KISSConfig,
```
Comment (nachovizzo) [inline src/python/kiss_icp/pipeline.py:44]: This change is in general desire, but I see the disadvantage of always having to build a `KISSConfig` object to use the pipeline. The other arguments are more "standard" which basically means I can fire up a python shell now, and `from kiss_icp.pipeline import Odometrypipeline` and run it as it it.

Would you mind reverting this back to the old bheavior?

File: src/python/kiss_icp/tools/cmd.py [hunk 2]
```diff
@@ -93,8 +93,8 @@ def kiss_icp_pipeline(
         callback=name_callback,
         help="[Optional] Use a specific dataloader from those supported by KISS-ICP",
     ),
-    config: Path = typer.Option(
-        f"{get_package_dir()}/config/default.yaml",
```
Comment (nachovizzo) [inline src/python/kiss_icp/tools/cmd.py:97]: `get_package_dir` function can also be removed

**Review by nachovizzo — CHANGES_REQUESTED** [review]
I think I end up requesting a lot. Please tell me if you can't solve all the issues, I will merge your changes and modify it afterwards.

Thanks again for your contributions!!!

File: setup.py [hunk 3]
```diff
@@ -68,10 +68,10 @@ def cmd_exists(cmd):
         package_data={"kiss_icp": ["config/default.yaml"]},
         install_requires=[
             "PyYAML",
-            "easydict",
             "natsort",
             "numpy",
             "plyfile",
+            "pydantic",
```
[the remark this answers is not in the document]
Comment (nachovizzo) [reply setup.py:74]: This is possible, but it's a pain. So, dropping this idea

**Comment by nachovizzo** [comment]
Merging these changes in #69

**Comment by markuspi** [comment]
Thanks for refining 👍

**Pull request closed without merging**